File: sdk/android/sqljson/controller/NoteController.java
package ardvro.component.sqljson.controller;

import com.google.gson.internal.LinkedTreeMap;

import java.util.ArrayList;

import ardvro.adapter.gen.genNote;
import ardvro.core.connection.IConnection;
import ardvro.core.connection.IResponseListener;

public class NoteController
{
    private IConnection _conection;

    public NoteController(IConnection connection)
    {
        _conection = connection;
    }

    public void Delete(String name, IResponseListener<genNote> onResponse)
    {
        String functionName = "ardvro/component/sqljson/note/delete";
        _conection.Submit(new Object[]{ name }, functionName, onResponse);
    }

    public void Get(String name, IResponseListener<genNote> onResponse)
    {
        String functionName = "ardvro/component/sqljson/note/Get";
        _conection.Submit(new Object[]{ name }, functionName, onResponse);
    }

    public void GetByUsername(IResponseListener<ArrayList<genNote>> onResponse)
    {
        String functionName = "ardvro/component/sqljson/note/Getbyusername";
        _conection.Submit(new Object[]{  }, functionName, onResponse);
    }

    public void Save(String name, String note, IResponseListener<genNote> onResponse)
    {
        LinkedTreeMap obj = new LinkedTreeMap();
        obj.put("Name", name);
        obj.put("Content", note);
        String functionName = "ardvro/component/sqljson/note/Get";
        _conection.Submit(obj, functionName, onResponse);
    }

}

File: sdk/.net/ardvro.sdk.net/component/sqljson/controller/NoteController.cs
namespace ardvro.sdk.net.component.sqljson.controller;

using ardvro.adapter.gen.ent;
using ardvro.core.ext.connection;
using ardvro.core.lib.common;

public class NoteController
{
    private IConnection _conection;

    public NoteController(IConnection connection)
    {
        _conection = connection;
    }

    public void Delete(String name, Action<genNote> onResponse)
    {
        String functionName = "ardvro/component/sqljson/note/delete";
        _conection.Submit(new Object[]{ name }, functionName, onResponse);
    }

    public void Get(String name, Action<genNote> onResponse)
    {
        String functionName = "ardvro/component/sqljson/note/Get";
        _conection.Submit(new Object[]{ name }, functionName, onResponse);
    }

    public void GetByUsername(Action<List<genNote>> onResponse)
    {
        String functionName = "ardvro/component/sqljson/note/Getbyusername";
        _conection.Submit(new Object[]{  }, functionName, onResponse);
    }

    public void Save(String name, String note, Action<genNote> onResponse)
    {
        var obj = new Flexible();
        obj.Add("Name", name);
        obj.Add("Content", note);
        String functionName = "ardvro/component/sqljson/note/Get";
        _conection.Submit(obj, functionName, onResponse);
    }

}
